### DISCUSSION TITLE
V6 - multiple step form one the same route

### DISCUSSION THREAD

**Discussion by krasimirkostadinov** [react-hook-form/react-hook-form#4742] [Q&A]
Hey guys, i reviewed many examples and tutorials. But my case is having one form with multiple steps and each step should got "isValid" from the subset fields for the current step. For example i'm having 15 fields on 3 step (each by 5 fields).

I'm working over this example: https://codesandbox.io/s/unruffled-river-9fkxo?file=/src/MultiStepForm.js:1419-1636

But 
**Should i use different forms on every step and reset it, and finally collect data from Redux store, or one form with partial form subset**? I don't have different view/components for every step (like step1.js, step2.jsx ect...) , cuz it'z dynamic. So on every <form> render i'm passing "currentStepFields" which is array[fields] for the current step until finally dispatch submit with collected data from redux store. 

This way i'm having problem with default values when switching forward -> backward from and reviewing steps and <input defaultValue is got  intertwine with other form fields. maybe ref={register} problem.

**Comment by bluebill1049** [comment]
This is a tricky one, I think it's better to manage per form/section. input gets unmounted shouldn't be validated again or taking count as valid or invalid inputs.
